rules/utils.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 Google LLC
```

Comment:
utils.go is pretty generic, and could become a huge dumping ground. Maybe name this file http_annotations.go to keep all utilities around annotations co-located?

Reply:
I have almost the opposite concern -- given that this is a `rules` package, the number of things that are not rules should be as limited as possible.

Reply:
Maybe you want to create an internal package for util: `internal/util` under `rules`.

Reply:
That makes me unable to read the `testdata`.